async_producer_test.go

```diff
@@ -320,7 +320,7 @@ func TestAsyncProducerEncoderFailures(t *testing.T) {
 	leader.Returns(prodSuccess)
 
 	config := NewConfig()
-	config.Producer.Flush.Messages = 3
+	config.Producer.Flush.Messages = 1
```

Comment:
This was needed because the 2 failed to encode messages are not counted towards the batch anymore I assume?